idaes/surrogate/pysmo/sampling.py

```diff
@@ -682,25 +773,31 @@ class HaltonSampling(SamplingMethods):
 
     """
 
-    def __init__(self, data_input, number_of_samples=None, sampling_type=None):
+    def __init__(self, data_input, number_of_samples=None, sampling_type=None, xlabels=None, ylabels=None):
         """
 
         Initialization of **HaltonSampling** class. Two inputs are required.
 
         Args:
             data_input (NumPy Array, Pandas Dataframe or list) : The input data set or range to be sampled.
             
-                - When the aim is to select a set of samples from an existing dataset, the dataset must be a NumPy Array or a Pandas Dataframe and **sampling_type** option must be set to "selection". The output variable (Y) is assumed to be supplied in the last column.
-                - When the aim is to generate a set of samples from a data range, the dataset must be a list containing two lists of equal lengths which contain the variable bounds and **sampling_type** option must be set to "creation". It is assumed that no range contains no output variable information  in this case.
-            
+                - When the aim is to select a set of samples from an existing dataset, the dataset must be a NumPy Array or a Pandas Dataframe and **sampling_type** option must be set to "selection". A single output variable (y) is assumed to be supplied in the last column if **xlabels** and **ylabels** are not supplied.
+                - When the aim is to generate a set of samples from a data range, the dataset must be a list containing two lists of equal lengths which contain the variable bounds and **sampling_type** option must be set to "creation". It is assumed that the range contains no output variable information  in this case.
+
             number_of_samples(int): The number of samples to be generated. Should be a positive integer less than or equal to the number of entries (rows) in **data_input**.
             sampling_type(str) : Option which determines whether the algorithm selects samples from an existing dataset ("selection") or attempts to generate sample from a supplied range ("creation"). Default is "creation".
 
+        Keyword Args:
+            xlabels (list): List of column names (if **data_input** is a dataframe) or column numbers (if **data_input** is an array) for the independent/input  variables.  Only used in "selection" mode. Default is None.
+            ylabels (list): List of column names (if **data_input** is a dataframe) or column numbers (if **data_input** is an array) for the dependent/output variables. Only used in "selection" mode. Default is None.
+            
         Returns:
             **self** function containing the input information.
 
         Raises:
             ValueError: The **data_input** is the wrong type.
+
+            IndexError: When invalid column names are supplied in **xlabels** or **ylabels**
```

Comment:
style nitpick: rather than **bold**, use _emphasis_ or `literal` (in restructured text syntax, using one or two backticks like \`emphasis\` or \`\`literal\`\`)  to emphasize strings that refer to variable or function names.